src/inmanta/server/services/environmentservice.py

```diff
@@ -269,6 +269,9 @@ async def halt(self, env: data.Environment) -> None:
 
     @handle(methods_v2.resume_environment, env="tid")
     async def resume(self, env: data.Environment) -> None:
+        if env.is_marked_for_deletion:
+            raise BadRequest("Cannot resume an environment that is marked for deletion.")
+
```

Comment:
As the `agent_state_lock` is already used for the halting/resuming I used the same one for the delete. I'm not sure this  is ok.

Reply:
Indeed. There is already a lock for the agent_state. The thing I don't like about re-using the same lock for the mark_for_deletion flag, is that it becomes very implicit that this lock is used for two purposes. I would either introduce a second lock or I would rename the current lock so that it reflects its broader scope and add a comment that explain the scope of the lock.